packages/@n8n/nodes-langchain/nodes/memory/MemoryManager/MemoryManager.node.ts

```diff
@@ -0,0 +1,329 @@
+										value: 'system',
+									},
+									{
+										name: 'User',
+										value: 'user',
+									},
+								],
+								default: 'system',
+							},
+							{
+								displayName: 'Message',
+								name: 'message',
+								type: 'string',
+								required: true,
+								default: '',
+							},
+							{
+								displayName: 'Hide Message in Chat',
+								name: 'hideFromUI',
+								type: 'boolean',
+								required: true,
+								default: false,
+								description: 'Whether to hide the message from the chat UI',
+							},
+						],
+					},
+				],
+				displayOptions: {
+					show: {
+						mode: ['insert'],
+					},
+				},
+			},
+			{
+				displayName: 'Messages Count',
+				name: 'lastMessagesCount',
+				type: 'number',
+				description: 'The amount of last messages to delete',
+				default: 2,
+				displayOptions: {
+					show: {
+						mode: ['delete'],
+						deleteMode: ['lastN'],
+					},
+				},
+			},
+			{
+				displayName: 'Simplify Output',
+				name: 'simplifyOutput',
+				type: 'boolean',
+				description: 'Whether to simplify the output to only include the sender and the text',
+				default: true,
+				displayOptions: {
+					show: {
+						mode: ['load'],
+					},
+				},
+			},
+		],
+	};
+
+	async execute(this: IExecuteFunctions): Promise<INodeExecutionData[][]> {
+		const memory = (await this.getInputConnectionData(
+			NodeConnectionType.AiMemory,
+			0,
+		)) as BaseChatMemory;
+
+		const items = this.getInputData();
+		const result = [];
+		for (let i = 0; i < items.length; i++) {
+			const mode = this.getNodeParameter('mode', i) as 'load' | 'insert' | 'delete';
+
+			let messages = [...(await memory.chatHistory.getMessages())];
+
+			if (mode === 'delete') {
+				const deleteMode = this.getNodeParameter('deleteMode', i) as 'lastN' | 'all';
+
+				if (deleteMode === 'lastN') {
+					const lastMessagesCount = this.getNodeParameter('lastMessagesCount', i) as number;
+					if (messages.length >= lastMessagesCount) {
+						const newMessages = messages.slice(0, messages.length - lastMessagesCount);
+
+						await memory.chatHistory.clear();
+						for (const message of newMessages) {
+							await memory.chatHistory.addMessage(message);
+						}
+					}
+				} else {
+					await memory.chatHistory.clear();
+				}
+			}
+
+			if (mode === 'insert') {
+				const insertMode = this.getNodeParameter('insertMode', i) as 'insert' | 'override';
+				const messagesToInsert = this.getNodeParameter(
+					'messages.messageValues',
+					i,
+					[],
+				) as MessageRecord[];
+
+				const templateMapper = {
+					ai: AIMessage,
+					system: SystemMessage,
+					user: HumanMessage,
+				};
+
+				if (insertMode === 'override') {
+					await memory.chatHistory.clear();
+				}
+
+				for (const message of messagesToInsert) {
+					const MessageClass = new templateMapper[message.type](message.message);
+
+					if (message.hideFromUI) {
+						MessageClass.additional_kwargs.hideFromUI = true;
```

Comment:
would `MessageClass` always have `additional_kwargs`? should we consider case when it's `undefined`?

Reply:
It should, as per Langchain types, this propert is not optional and `NonNullable`:

```ts
/**
 * Exclude null and undefined from T
 */
type NonNullable<T> = T & {};

    /** Additional keyword arguments */
    additional_kwargs: NonNullable<BaseMessageFields["additional_kwargs"]>;
```